src/vcRender.cpp

```diff
@@ -1148,6 +1161,11 @@ void vcRender_RenderScene(vcState *pProgramState, vcRenderContext *pRenderContex
 
   if (pProgramState->settings.presentation.mouseAnchor != vcAS_None && (pProgramState->pickingSuccess || pProgramState->isUsingAnchorPoint))
   {
+    // resolve (last frame) polygon vs. (current frame) UD here
+    udDouble3 pickPosition = vcRender_DepthToWorldPosition(pProgramState, pRenderContext, pRenderContext->previousFrameDepth);
+    if (udMagSq(pickPosition - pProgramState->camera.position) < udMagSq(pProgramState->worldMousePosCartesian - pProgramState->camera.position))
+      pProgramState->worldMousePosCartesian = pickPosition;
```

Comment:
Isn't this logic currently living in main.cpp to resolve between different depth sources?